src/OpenLoco/src/GameCommands/ChangeLandMaterial.cpp

```diff
@@ -27,6 +27,9 @@ namespace OpenLoco::GameCommands
         World::TilePosRangeView tileLoop{ World::toTileSpace(pointA), World::toTileSpace(pointB) };
         for (const auto& tilePos : tileLoop)
         {
+            if (!World::validCoords(tilePos))
+                continue;
```

Comment:
I'm honestly not a fan having this in loops, wouldn't it be better to ensure that that the range specified is correct? It's probably better to have a function that clamps TilePosRangeView to the map bounds.

Reply:
I agree, this is clearly very error-prone. However, when previously discussed it wasn't an option for `TilePosRangeView` to be constrained in this way? I don't quite recall. I would prefer the range view to handle this as well, though.

Reply:
Yeah that could work

Reply:
I've pushed a suggestion introducing `getClampedRange` and `getWorldRange` as a means of obtaining a `TilePosRangeView`. Still needs testing.